e2e_playwright/st_set_page_config_test.py

```diff
@@ -47,6 +47,33 @@ def test_wide_layout(app: Page):
     assert narrow_expander_width < expander_dimensions["width"]
 
 
+def test_wide_layout_with_small_viewport(app: Page):
+    """Test that the wide layout is using the same width as the centered layout
+    when the viewport is narrow."""
+
+    app.set_viewport_size({"width": 640, "height": 800})
+
+    app_view_container = app.get_by_test_id("stAppViewContainer")
+    # The default layout is "centered":
+    expect(app_view_container).to_have_attribute("data-layout", "narrow")
+
+    expander_container = get_expander(app, "Expander in main")
+    expect(expander_container).to_be_visible()
+    expander_dimensions = expander_container.bounding_box()
+    assert expander_dimensions is not None
```

Comment:
**suggestion (non-blocking):** I would expect tests to leverage the same style of assertions unless there are specific behaviors we want to leverage with `assert` vs `expect`. My preference would be that we leverage `expect` everywhere, but it seems like we already have a number of mixed usages in our tests today, so this is not blocking feedback for this PR.

Reply:
agree on keeping it unified. But I think there is no `expect` from playwright that allows us to do this kind of value check (see [here](https://playwright.dev/python/docs/test-assertions)). In general, `assert` should never be used in playwright tests unless the developer is certain that the check cannot cause any flakiness. In case its not certain that the value is filled with the correct value immediately, we have the internal `wait_until` method that imitates the `expect` behaviour from playwright: see [this](https://github.com/streamlit/streamlit/blob/bb99ba7072f6689dc5ab084ec894dca681a59786/e2e_playwright/hostframe_app_test.py#L240-L244). We could use it here as well, but its a bit unnecessary since the `expander_dimensions` is guaranteed to filled immediately after the element becomes visible.

Reply:
btw. the two `assert expander_dimensions is not None` are only for mypy -> I think I could just replace it with type ignores. But the more relevant assert is the one in line 74: 

```python
assert narrow_expander_width == expander_dimensions["width"]
```

Reply:
That makes sense, given the use case `assert` is probably better than mypy type ignores!